pkg/gadgets/trace/bind/tracer/tracer_test.go

```diff
@@ -0,0 +1,649 @@
+					opt:   unix.IP_BIND_ADDRESS_NO_PORT,
+					value: 1,
+				},
+				{
+					level: unix.IPPROTO_IP,
+					opt:   unix.IP_FREEBIND,
+					value: 1,
+				},
+			},
+			expectedOpts: "..NTF",
+		},
+	} {
+		test := test
+
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			events := []types.Event{}
+			eventCallback := func(event types.Event) {
+				events = append(events, event)
+			}
+
+			runner := utilstest.NewRunnerWithTest(t, nil)
+
+			tracerConfig := &tracer.Config{
+				MountnsMap: utilstest.CreateMntNsFilterMap(t, runner.Info.MountNsID),
+			}
+
+			createTracer(t, tracerConfig, eventCallback)
+
+			utilstest.RunWithRunner(t, runner, func() error {
+				_, err := bindSocketWithOpts("127.0.0.1", unix.AF_INET, unix.SOCK_STREAM, 0, test.opts)
+				return err
+			})
+
+			// Give some time for the tracer to capture the events
+			time.Sleep(100 * time.Millisecond)
+
+			if len(events) != 1 {
+				t.Fatalf("Wrong number of events received %d, expected 1", len(events))
+			}
+
+			if events[0].Options != test.expectedOpts {
+				t.Fatalf("Wrong options received: %s, expected %s",
+					events[0].Options, test.expectedOpts)
+			}
+		})
+	}
+}
+
+func TestBindTracerMultipleMntNsIDsFilter(t *testing.T) {
+	t.Parallel()
+
+	utilstest.RequireRoot(t)
+
+	events := []types.Event{}
+	eventCallback := func(event types.Event) {
+		events = append(events, event)
+	}
+
+	// struct with only fields we want to check on this test
+	type expectedEvent struct {
+		mntNsID uint64
+		port    uint16
+	}
+
+	const n int = 5
+	runners := make([]*utilstest.Runner, n)
+	expectedEvents := make([]expectedEvent, n)
+	mntNsIDs := make([]uint64, n)
+
+	for i := 0; i < n; i++ {
+		runners[i] = utilstest.NewRunnerWithTest(t, nil)
+		mntNsIDs[i] = runners[i].Info.MountNsID
+		expectedEvents[i].mntNsID = runners[i].Info.MountNsID
+	}
+
+	// Filter events from all runners but last one
+	config := &tracer.Config{
+		MountnsMap: utilstest.CreateMntNsFilterMap(t, mntNsIDs[:n-1]...),
+	}
+
+	createTracer(t, config, eventCallback)
+
+	for i := 0; i < n; i++ {
+		utilstest.RunWithRunner(t, runners[i], func() error {
+			var err error
+			expectedEvents[i].port, err = bindSocket("127.0.0.1", unix.AF_INET, unix.SOCK_STREAM, 0)
+			return err
+		})
+	}
+
+	// Give some time for the tracer to capture the events
+	time.Sleep(100 * time.Millisecond)
+
+	if len(events) != n-1 {
+		t.Fatalf("%d events were expected, %d found", n, len(events))
+	}
+
+	// Pop last event since it shouldn't have been captured
+	expectedEvents = expectedEvents[:n-1]
+
+	// Order or events is not guaranteed, then we need to sort before comparing
+	sort.Slice(expectedEvents, func(i, j int) bool {
+		return expectedEvents[i].mntNsID < expectedEvents[j].mntNsID
+	})
+	sort.Slice(events, func(i, j int) bool {
+		return events[i].MountNsID < events[j].MountNsID
+	})
+
+	for i := 0; i < n-1; i++ {
+		if events[i].MountNsID != expectedEvents[i].mntNsID {
+			t.Fatalf("Captured event has bad MountNsID")
```

Comment:
Why not print the values?

Reply:
I introduced a new helper to make this kind of comparison easier.